internal/server/health_check.go

```diff
@@ -94,7 +95,11 @@ func (hc *HealthCheck) check() {
 		hc.reportResult(false, err)
 		return
 	}
-	defer resp.Body.Close()
+
+	defer func() {
+		_, _ = io.Copy(io.Discard, resp.Body)
+		resp.Body.Close()
+	}()
```

Comment:
Minor thing, but I think discarding the body immediately, rather than deferring it, is a little more readable and idiomatic:

```suggestion
        defer resp.Body.Close()
        
        _, _ = io.Copy(io.Discard, resp.Body)
```